### ISSUE TITLE
static checker is not testing against the current PR?

### ISSUE THREAD

**Issue by bergwolf** [kata-containers/tests#4388]
It passed in https://github.com/kata-containers/kata-containers/pull/3468 but resulted in various failures after kata-containers/kata-containers#3468 was merged.

E.g.,
https://github.com/kata-containers/kata-containers/runs/4876000763?check_suite_focus=true
https://github.com/kata-containers/kata-containers/runs/4876769522?check_suite_focus=true
https://github.com/kata-containers/kata-containers/runs/4876343215?check_suite_focus=true

**Comment by bergwolf** [comment]
Another evidence that the static checker is not testing the current PR. This one does not have `force-skip-ci` label. Seeing after https://github.com/kata-containers/kata-containers/pull/3494 was merged:

20T01:28:21.951721646Z" level=info msg="Checking file" commit=ce03e521b01a4ad14f91ce9118c6e6de16077839 file=tools/packaging/README.md name=kata-check-markdown pid=4061722 source=check-markdown version=0.0.1
01:28:21 time="2022-01-20T01:28:21.955502462Z" level=fatal msg="found 1 parse error:\nstat tools/packaging/cmd/kata-pkgsync: no such file or directory"

**Comment by fidencio** [comment]
Let me speak about https://github.com/kata-containers/kata-containers/pull/3468.

On https://github.com/kata-containers/kata-containers/pull/3468 I've brought in the GoVMM into the Kata Containers repo, and I have used the `force-skip-ci` label to bring that in.  The reason for that was because the checks on the commit shortlog subsystem simply didn´t pass our CI, and I preferred to keep the whole history of the project.

With that said, I went ahead, set the `force-skip-ci` and started working on fixes before the #3468 got merged, and I did it on my own fork.  Once I got everything passed on my own fork, I then merged #3468 and opened #3496.

On #3496, though, I noticed issues related to the s390x CI, which where then tracked on #3500 ... and once #3496 gets merged things are unblocked.  I am re-running some of the CIs as those are failing with errors not related to that and that's it.

I am sorry for the breakage here.

**Comment by fidencio** [comment]
About https://github.com/kata-containers/tests/issues/4388#issuecomment-1017072070, it **looks like** the submitter should rebase the PR.

**Comment by fidencio** [comment]
> About [#4388 (comment)](https://github.com/kata-containers/tests/issues/4388#issuecomment-1017072070), it **looks like** the submitter should rebase the PR.

I just checked https://github.com/kata-containers/kata-containers/runs/4879521064?check_suite_focus=true, with https://github.com/kata-containers/kata-containers/pull/3483 rebased atop of main and it does work.

So, here what we should be focusing on is ensuring that we do such a rebase when checking out the code and running the static checks.

**Comment by bergwolf** [comment]
Thanks for the analysis @fidencio ! But I think https://github.com/kata-containers/tests/issues/4388#issuecomment-1017072070 is more than just a rebase. Let's recap what happened:
1. https://github.com/kata-containers/kata-containers/pull/3494 was merged and it removed `kata-pkgsync`
2. Another PR that was rejected because the static checker cannot find `kata-pkgsync`, which means it includes the changes made by https://github.com/kata-containers/kata-containers/pull/3494

I don't think we should focus on 2) on whether it needs to rebase or not. But instead, we should focus on  https://github.com/kata-containers/kata-containers/pull/3494 why it passed the static checker in the first place. It removed `kata-pkgsync`, in theory, it should have removed all the references in markdown to `kata-pkgsync` otherwise static checker should fail.

But now https://github.com/kata-containers/kata-containers/pull/3494 was merged and later another PR failed because static-checker cannot find `kata-pkgsync`, which is a sign that the reference was not removed in https://github.com/kata-containers/kata-containers/pull/3494. Looking at the commit history, I found that the reference to `kata-pkgsync` was removed in another PR https://github.com/kata-containers/kata-containers/pull/3514. If the static checker had worked properly, it should have gated https://github.com/kata-containers/kata-containers/pull/3494 and forced the PR author to fold changes of https://github.com/kata-containers/kata-containers/pull/3514 to https://github.com/kata-containers/kata-containers/pull/3494. That is why I think the static checker is not testing against the current PR, or at least it is missing pieces that should have been checked.

**Comment by fidencio** [comment]
Yep, I agree, @bergwolf.

The static-checks are not doing a reasonably enough job, not even to fail consistently.
